Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -337,7 +338,11 @@ class BrowserViewController: UIViewController {
 
             addChildViewController(homePanelController!)
         }
-        homePanelController?.selectedButtonIndex = tabManager.selectedTab?.lastHomePanel
+
+        var panelNumber = tabManager.selectedTab?.url?.fragment
+        var numberArray = panelNumber?.componentsSeparatedByString("=")
+        homePanelController?.selectedButtonIndex = numberArray?.last?.toInt()
```

Comment:
Let's add `?? 0` to the end so that we fall back to the first panel as a default.